scheduler/core/components/collector/collector.py

```diff
@@ -380,7 +380,7 @@ def _calculate_target_info(self,
             # Are all the required resources available?
             # This works for validation mode. In RT mode, this may need to be statistical if resources are not known
             # and they could change with time, so the visfrac calc may need to be extracted from this method
-            has_resources = all([resource in nc[night_idx].resources for resource in obs.required_resources()])
+            has_resources = all([resource.id in [r.id for r in nc[night_idx].resources] for resource in obs.required_resources()])
```

Comment:
It didn't want to copy my new chat log, so here are the addendums. This will require moving `ResourceManager` to `lucupy`, but I see no reason not to, since it isn't specific to the Scheduler and could be used anywhere.

<img width="623" alt="Screenshot 2024-03-20 at 2 10 17 PM" src="https://github.com/gemini-hlsw/scheduler/assets/8795653/8b52e33b-11d2-4aa9-abf0-8006a6e8f67b">
<img width="622" alt="Screenshot 2024-03-20 at 2 10 31 PM" src="https://github.com/gemini-hlsw/scheduler/assets/8795653/cb91c247-4631-4055-853f-855d359288b6">

Reply:
Also, even in the original code, I'm not sure why a generator isn't being used. If we need to go with this implementation (which raises red flags for me), it should be:

```
night_resources_ids = frozenset({r.id for r in nc[night_idx].resources})
has_resources = all(resource.id in night_resources_ids for resource in obs.required_resources())
```